robot_smach_states/tutorial/challenge_tutorial.md

```diff
@@ -0,0 +1,236 @@
+# Creating a TU/e RoboCup challenge
+
+This tutorial explains how to setup a new RoboCup challenge. It assumes that you have finished the [ROS tutorials](http://wiki.ros.org/ROS/Tutorials) as well as the [Smach tutorials](http://wiki.ros.org/smach/Tutorials). Furthermore, it assumes that the simulator as well as 'middleware' are running. (ToDo: add links.)
+
+## Setting up the boiler plate
+
+### Setting up the package
+
+* Browse to the `tue_robocup` folder on your machine (ToDo: either create it directly in workspace, create link or whole target meuk):
+```
+cd ~/ros/$ROS_DISTRO/repos/https_/github.com/tue-robotics/tue_robocup.git/
+```
+* Create the package:
+```
+catkin_create_pkg challenge_example robot_skills robot_smach_states
+```
+This creates a ROS package called `challenge_example`. It depends on the `robot_skills`
+and the `robot_smach_states` package. This can be seen when inspecting the `package.xml` that is located inside your newly created `challenge_example` folder.
+* Make and source your `~/.bashrc` to make sure the package is in your `ROS_PACKAGE_PATH`.:
+```
+tue-make
+source ~/.bashrc
+```
+and go to your newly created package:
+```
+roscd challenge_example
+```
+to see if everything has worked.
+
+### Create the ROS node
+
+* In the `challenge_example` package, create a new `scripts` folder and enter it:
+```
+mkdir scripts
+cd scripts
+```
+* Create you new node and make it executable:
+```
+touch challenge_example
+chmod +x challenge_example
+```
+Using `ls` in your terminal you can see that the file is green.
+* Open your file in Pycharm and add the following contents:
```

Comment:
How to open file with pycharm?

Reply:
`pycharm-community challenge_example` worked for me, but you also don't have to open it in pycharm, for example sublime worked too.